pkg/snapstore/s3_snapstore.go

```diff
@@ -733,15 +759,23 @@ func GetS3CredentialsLastModifiedTime() (time.Time, error) {
 
 	if dir, isSet := os.LookupEnv(awsCredentialDirectory); isSet {
 		// credential files which are essential for creating the S3 snapstore
-		credentialFiles := []string{"accessKeyID", "region", "secretAccessKey"}
-		for i := range credentialFiles {
-			credentialFiles[i] = filepath.Join(dir, credentialFiles[i])
+		// either static or web identity credential files should be set
+		var (
+			staticCredentialsFiles      = []string{filepath.Join(dir, "region"), filepath.Join(dir, "accessKeyID"), filepath.Join(dir, "secretAccessKey")}
+			webIdentityCredentialsFiles = []string{filepath.Join(dir, "region"), filepath.Join(dir, "roleARN"), filepath.Join(dir, "token")}
+		)
+
+		awsTimestamp, err := getLatestCredentialsModifiedTime(staticCredentialsFiles)
```

Comment:
```suggestion
		awsTimestamp, err1 := getLatestCredentialsModifiedTime(staticCredentialsFiles)
```
you're using `err2` in next line, either change that or change this ... no strong opinion